### ISSUE TITLE
Data load failing when the measure data is null

### ISSUE THREAD

**Issue by foryou2030** [HuaweiBigData/carbondata#328]
When i use carbon to load data, if the measure columns of data csv is "", such as:
          the cube schema is:
             DIMENSIONS (name string, age integer) MEASURES (grade numeric)
          and data csv is:
             jacky,36,

i got an exception:

```
 16/05/06 11:54:17 ERROR step.CarbonSliceMergerStep: [vardhan: Graph - Carbon Slice Mergervardhan][partitionID:vardhan] {0}
org.pentaho.di.core.exception.KettleException: 
org.carbondata.processing.merger.exeception.SliceMergerException: Problem while renaming inprogress folder to actual
Problem while renaming inprogress folder to actual

	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.renameFolders(CarbonSliceMergerStep.java:170)
	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.processRow(CarbonSliceMergerStep.java:119)
	at org.pentaho.di.trans.step.RunThread.run(RunThread.java:40)
	at java.lang.Thread.run(Thread.java:722)
Caused by: org.carbondata.processing.merger.exeception.SliceMergerException: Problem while renaming inprogress folder to actual
	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.renameLoadFolderFromInProgressToNormal(CarbonSliceMergerStep.java:234)
	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.renameFolders(CarbonSliceMergerStep.java:163)
	... 3 more
16/05/06 11:54:17 INFO csvload.DataGraphExecuter: [Executor task launch worker-1][partitionID:default_vardhan_d538894c-7794-4ae7-8e6a-4dc4cb50772c] Graph execution is finished.
16/05/06 11:54:17 ERROR csvload.DataGraphExecuter: [Executor task launch worker-1][partitionID:default_vardhan_d538894c-7794-4ae7-8e6a-4dc4cb50772c] Graph Execution had errors
16/05/06 11:54:17 ERROR rdd.CarbonDataLoadRDD: [Executor task launch worker-1][partitionID:default_vardhan_d538894c-7794-4ae7-8e6a-4dc4cb50772c] {0}
org.carbondata.processing.etl.DataLoadingException: Internal Errors
	at org.carbondata.processing.csvload.DataGraphExecuter.execute(DataGraphExecuter.java:263)
	at org.carbondata.processing.csvload.DataGraphExecuter.executeGraph(DataGraphExecuter.java:174)
	at org.carbondata.integration.spark.load.CarbonLoaderUtil.executeGraph(CarbonLoaderUtil.java:187)
	at org.carbondata.integration.spark.rdd.CarbonDataLoadRDD$$anon$1.<init>(CarbonDataLoadRDD.scala:220)
	at org.carbondata.integration.spark.rdd.CarbonDataLoadRDD.compute(CarbonDataLoadRDD.scala:160)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:300)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:264)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
	at org.apache.spark.scheduler.Task.run(Task.scala:88)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:214)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:722)
```

@sraghunandan , pls check it.

When i verify smartcare scenario, the fact datas contain "" measures.

create statement:

`CREATE CUBE smartcare DIMENSIONS(BEGIN_TIME STRING, END_TIME STRING, MSISDN STRING, IMSI STRING, IMEI STRING, SID STRING, PROBEID STRING, INTERFACEID STRING, GROUPID STRING, GGSN_ID STRING, SGSN_ID STRING, SESSION_INDICATOR STRING, BEGIN_TIME_MSEL STRING, END_TIME_MSEL STRING, PROT_CATEGORY Integer, PROT_TYPE Integer, L7_CARRIER_PROT Integer, ENCRYPT_VERSION Integer, ROAMING_TYPE Integer, ROAM_DIRECTION Integer, MS_IP STRING, SERVER_IP STRING, MS_PORT Integer, SERVER_PORT Integer, APN STRING, SGSN_SIG_IP STRING, GGSN_SIG_IP STRING, SGSN_USER_IP STRING, GGSN_USER_IP STRING, MCC STRING, MNC STRING, RAT Integer, LAC STRING, RAC STRING, SAC STRING, CI STRING, BROWSER_TYPE Integer, HOST STRING, STREAMING_URL STRING, VIDEO_CODEC_ID STRING, AUDIO_CODEC_ID STRING, TAC STRING, ECI STRING, RAN_NE_USER_IP STRING, HOMEMCC STRING, HOMEMNC STRING, ID STRING) MEASURES(L4_UL_THROUGHPUT Numeric, L4_DW_THROUGHPUT Numeric, L4_UL_GOODPUT Numeric, L4_DW_GOODPUT Numeric, NETWORK_UL_TRAFFIC Numeric, NETWORK_DL_TRAFFIC Numeric, L4_UL_PACKETS Numeric, L4_DW_PACKETS Numeric, TCP_CONN_STATES Numeric, TCP_STATES Numeric, TCP_RTT Numeric, TCP_UL_OUTOFSEQU Numeric, TCP_DW_OUTOFSEQU Numeric, TCP_UL_RETRANS Numeric, TCP_DW_RETRANS Numeric, TCP_WIN_SIZE Numeric, TCP_MSS Numeric, TCP_CONN_TIMES Numeric, TCP_CONN_2_FAILED_TIMES Numeric, TCP_CONN_3_FAILED_TIMES Numeric, GET_STREAMING_FAILED_CODE Numeric, GET_STREAMING_FLAG Numeric, GET_STREAMING_DELAY Numeric, GET_NUM Numeric, GET_SUCCEED_NUM Numeric, GET_RETRANS_NUM Numeric, GET_TIMEOUT_NUM Numeric, GET_MAX_UL_SIZE Numeric, GET_MIN_UL_SIZE Numeric, GET_MAX_DL_SIZE Numeric, GET_MIN_DL_SIZE Numeric, INTBUFFER_FST_FLAG Numeric, INTBUFFER_FULL_FLAG Numeric, INTBUFFER_FULL_DELAY Numeric, STALL_NUM Numeric, STALL_DURATION Numeric, STREAMING_DW_PACKETS Numeric, STREAMING_DOWNLOAD_DELAY Numeric, PLAY_DURATION Numeric, STREAMING_QUALITY Numeric, VIDEO_DATA_RATE Numeric, VIDEO_FRAME_RATE Numeric, VIDEO_WIDTH Numeric, VIDEO_HEIGHT Numeric, AUDIO_DATA_RATE Numeric, STREAMING_FILESIZE Numeric, STREAMING_DURATIOIN Numeric, MEDIA_FILE_TYPE Numeric, PLAY_STATE Numeric, STREAMING_FLAG Numeric, TCP_STATUS_INDICATOR Numeric, DISCONNECTION_FLAG Numeric, FAILURE_CODE Numeric, FLAG Numeric, FLOW_SAMPLE_RATIO Numeric, TCP_SYN_TIME Numeric, TCP_SYN_TIME_MSEL Numeric, TCP_RTT_STEP1 Numeric, TCP_UL_RETRANS_WITHPL Numeric, TCP_DW_RETRANS_WITHPL Numeric, TCP_UL_RETRANS_WITHOUTPL Numeric, TCP_DW_RETRANS_WITHOUTPL Numeric, TCP_UL_PACKAGES_WITHPL Numeric, TCP_DW_PACKAGES_WITHPL Numeric, TCP_UL_PACKAGES_WITHOUTPL Numeric, TCP_DW_PACKAGES_WITHOUTPL Numeric, UPPERLAYER_IP_UL_PACKETS Numeric, UPPERLAYER_IP_DL_PACKETS Numeric, DOWNLAYER_IP_UL_PACKETS Numeric, DOWNLAYER_IP_DL_PACKETS Numeric, UPPERLAYER_IP_UL_FRAGMENTS Numeric, UPPERLAYER_IP_DL_FRAGMENTS Numeric, DOWNLAYER_IP_UL_FRAGMENTS Numeric, DOWNLAYER_IP_DL_FRAGMENTS Numeric, SERVICE_DALAY_MSEC Numeric, VALID_TRANS_DURATION Numeric, AIR_PORT_DURATION Numeric, RADIO_CONN_TIMES Numeric, TCP_FST_SYN_DIRECTION Numeric, RAN_NE_ID Numeric, HOMEPROVINCE Numeric, HOMECITY Numeric, SERVICE_ID Numeric, CHARGING_CLASS Numeric, CHARGE_FLAG Numeric, Prepaid_Flag Numeric) options(PARTITIONER [PARTITION_COUNT=1])`

load statement:

`LOAD DATA FACT FROM 'D:/hadoop/srcdata/smartcare.csv' INTO CUBE smartcare OPTIONS (DELIMITER ',', QUOTECHAR '"', FILEHEADER 'ID,SID,PROBEID,INTERFACEID,GROUPID,GGSN_ID,SGSN_ID,SESSION_INDICATOR,BEGIN_TIME,BEGIN_TIME_MSEL,END_TIME,END_TIME_MSEL,PROT_CATEGORY,PROT_TYPE,L7_CARRIER_PROT,MSISDN,IMSI,IMEI,ENCRYPT_VERSION,ROAMING_TYPE,ROAM_DIRECTION,MS_IP,SERVER_IP,MS_PORT,SERVER_PORT,APN,SGSN_SIG_IP,GGSN_SIG_IP,SGSN_USER_IP,GGSN_USER_IP,MCC,MNC,RAT,LAC,RAC,SAC,CI,BROWSER_TYPE,L4_UL_THROUGHPUT,L4_DW_THROUGHPUT,L4_UL_GOODPUT,L4_DW_GOODPUT,NETWORK_UL_TRAFFIC,NETWORK_DL_TRAFFIC,L4_UL_PACKETS,L4_DW_PACKETS,TCP_CONN_STATES,TCP_STATES,TCP_RTT,TCP_UL_OUTOFSEQU,TCP_DW_OUTOFSEQU,TCP_UL_RETRANS,TCP_DW_RETRANS,TCP_WIN_SIZE,TCP_MSS,TCP_CONN_TIMES,TCP_CONN_2_FAILED_TIMES,TCP_CONN_3_FAILED_TIMES,HOST,STREAMING_URL,GET_STREAMING_FAILED_CODE,GET_STREAMING_FLAG,GET_STREAMING_DELAY,GET_NUM,GET_SUCCEED_NUM,GET_RETRANS_NUM,GET_TIMEOUT_NUM,GET_MAX_UL_SIZE,GET_MIN_UL_SIZE,GET_MAX_DL_SIZE,GET_MIN_DL_SIZE,INTBUFFER_FST_FLAG,INTBUFFER_FULL_FLAG,INTBUFFER_FULL_DELAY,STALL_NUM,STALL_DURATION,STREAMING_DW_PACKETS,STREAMING_DOWNLOAD_DELAY,PLAY_DURATION,STREAMING_QUALITY,VIDEO_DATA_RATE,VIDEO_FRAME_RATE,VIDEO_CODEC_ID,VIDEO_WIDTH,VIDEO_HEIGHT,AUDIO_DATA_RATE,AUDIO_CODEC_ID,STREAMING_FILESIZE,STREAMING_DURATIOIN,MEDIA_FILE_TYPE,PLAY_STATE,STREAMING_FLAG,TCP_STATUS_INDICATOR,DISCONNECTION_FLAG,FAILURE_CODE,FLAG,FLOW_SAMPLE_RATIO,TAC,ECI,TCP_SYN_TIME,TCP_SYN_TIME_MSEL,TCP_RTT_STEP1,TCP_UL_RETRANS_WITHPL,TCP_DW_RETRANS_WITHPL,TCP_UL_RETRANS_WITHOUTPL,TCP_DW_RETRANS_WITHOUTPL,TCP_UL_PACKAGES_WITHPL,TCP_DW_PACKAGES_WITHPL,TCP_UL_PACKAGES_WITHOUTPL,TCP_DW_PACKAGES_WITHOUTPL,UPPERLAYER_IP_UL_PACKETS,UPPERLAYER_IP_DL_PACKETS,DOWNLAYER_IP_UL_PACKETS,DOWNLAYER_IP_DL_PACKETS,UPPERLAYER_IP_UL_FRAGMENTS,UPPERLAYER_IP_DL_FRAGMENTS,DOWNLAYER_IP_UL_FRAGMENTS,DOWNLAYER_IP_DL_FRAGMENTS,SERVICE_DALAY_MSEC,VALID_TRANS_DURATION,AIR_PORT_DURATION,RADIO_CONN_TIMES,TCP_FST_SYN_DIRECTION,RAN_NE_ID,RAN_NE_USER_IP,HOMEMCC,HOMEMNC,HOMEPROVINCE,HOMECITY,SERVICE_ID,CHARGING_CLASS,CHARGE_FLAG,Prepaid_Flag')`

fact data (smartcare.csv , one record)

232240006911370409385231,51626,6541,1,,9,3,2,1370409385,231,1370409385,538,6,342,0,19607411022,460003107411022,350027277411022,,,,172.168.1.72,172.168.1.108,92,119,HUAWEIAPN32,10.10.181.72,90.12.11.9,10.10.183.42,10.10.182.13,460,00,1,5A16,,2020,,35,13582784,108561268,181,195,13582784,108561268,3621,4968,0,1,0,16,37,45,27,3,1500,3,1,2,WWW.HUAWEI4.COM,WWW.HUAWEI123.COM,,,906,,,,,,,,,0,0,986,0,250,108561268,9362,127352,443,832,12,8,20,949,643,3202,78584187,1359,0,2,1,,1,135,0,1000,,,1370409385,0,458,1,2,1,1,22,156,88,159,225,217,21,28,15,4,9,1,279343075,443306271,7098479,8,0,2420,10.10.181.156,460,00,0,0,0,0,0,0,0

**Comment by sraghunandan** [comment]
@foryou2030 pls check if the issue is already resolved.If yes, please close the PR and the issue

**Comment by foryou2030** [comment]
@sraghunandan , It is still not working. the same error:

```
16/05/17 15:34:48 ERROR step.CarbonSliceMergerStep: [t3: Graph - Carbon Slice Mergert3][partitionID:t3] 
org.pentaho.di.core.exception.KettleException: 
org.carbondata.processing.merger.exeception.SliceMergerException: Problem while renaming inprogress folder to actual
Problem while renaming inprogress folder to actual

	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.renameFolders(CarbonSliceMergerStep.java:168)
	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.processRow(CarbonSliceMergerStep.java:118)
	at org.pentaho.di.trans.step.RunThread.run(RunThread.java:40)
	at java.lang.Thread.run(Thread.java:722)
Caused by: org.carbondata.processing.merger.exeception.SliceMergerException: Problem while renaming inprogress folder to actual
	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.renameLoadFolderFromInProgressToNormal(CarbonSliceMergerStep.java:232)
	at org.carbondata.processing.merger.step.CarbonSliceMergerStep.renameFolders(CarbonSliceMergerStep.java:161)
	... 3 more
```

**Comment by gvramana** [comment]
merged PR #425
